include/zephyr/net/dns_resolve.h

```diff
@@ -266,18 +279,59 @@ int dns_dispatcher_unregister(struct dns_socket_dispatcher *ctx);
 
 /** @endcond */
 
+/**
+ * Enumerate the extensions that are available in the address info
+ */
+enum dns_resolve_extension {
+	DNS_RESOLVE_NONE = 0,
+	DNS_RESOLVE_TXT,
+	DNS_RESOLVE_SRV,
+};
+
+struct dns_resolve_txt {
+	size_t textlen;
+	char   text[DNS_MAX_TEXT_SIZE + 1];
+};
+
+struct dns_resolve_srv {
+	uint16_t priority;
+	uint16_t weight;
+	uint16_t port;
+	size_t   targetlen;
+	char     target[DNS_MAX_NAME_SIZE + 1];
+};
+
 /**
  * Address info struct is passed to callback that gets all the results.
  */
 struct dns_addrinfo {
-	/** IP address information */
-	struct sockaddr ai_addr;
-	/** Length of the ai_addr field */
-	socklen_t       ai_addrlen;
-	/** Address family of the address information */
-	uint8_t         ai_family;
-	/** Canonical name of the address */
-	char            ai_canonname[DNS_MAX_NAME_SIZE + 1];
+	/** Address family of the address information and discriminator */
+	uint8_t ai_family;
+
+	union {
+		struct {
+			/** Length of the ai_addr field or ai_canonname */
+			socklen_t ai_addrlen;
+
+			union {
+				/* AF_INET or AF_INET6 address info */
+				struct sockaddr ai_addr;
```

Comment:
I'm not sure if `ai_addr` should be in union with `ai_canonname` - I think in theory both fields could be used with A/AAAA if `AI_CANONNAME` flag was used (which we don't support at the moment)?